matrix_client/api.py

```diff
@@ -134,6 +134,8 @@ def register(self, content={}, kind='user'):
 
             kind (str): Specify kind="guest" to register as guest.
         """
+        if content is None:
+            content = {}
```

Comment:
I'm confused, it causes havoc how? And as far as I can tell the outcome is exactly the same here.

Reply:
@pcppcp is this code doing gevent with multiple clients public anywhere by chance? Would be curious to see it.

Reply:
we use matrix as an experimental transport layer in [Raiden](https://github.com/raiden-network/raiden) and since in the tests there are sometimes many clients running in a single thread, a lot of context switching occurs and this had revealed the problem with the default dict.

Reply:
very cool

Reply:
@non-Jedi we plan to upstream some of our changes in the near future, but if you're interested on some concepts of our usage, [you may look at some of our usages here](https://github.com/raiden-network/raiden-libs/blob/master/raiden_libs/network/matrix/client.py#L17)